src/lib/gsheets/index.js

```diff
@@ -211,14 +225,39 @@ function gsheets(server) {
       updatedAtLabel,
     };
   }
+  async function fetchArticlesCategories() {
+    const categories = (await fetchCategories()).reduce(reduceByName, {});
+    const categoriesSheet = await fetchSheet(articlesCategoriesSheetId);
+    return categoriesSheet
+      .filter((row) => row.article && row.category)
+      .map(({ article, category }) => {
+        const id = extractIdFromEntity(article);
+        return {
+          ...categories[camelCase(category)],
+          article: { id },
+        };
+      })
+      .filter((row) => row.article.id);
+  }
 
   async function fetchArticles(cache) {
     const { data: site } = await cache.site;
     const articlesSheet = await fetchSheet(articlesSheetId);
+    const articleCategories = await fetchArticlesCategories();
 
     return articlesSheet
       .filter((row) => equalsIgnoreCase(row.site, site.slug))
-      .map(({ photo, ...others }) => ({ ...others, photo, image: photo }));
+      .map(({ photo, ...others }) => ({
+        ...others,
+        photo,
+        categories: articleCategories.flatMap(({ article, ...category }) => {
+          if (equalsIgnoreCase(article.id, others.id)) {
+            return [category];
+          }
+          return [];
+        }),
```

Comment:
```suggestion
        categories: articleCategories
            .filter(({ article }) => equalsIgnoreCase(article.id, others.id))
            .map(({ category }) => category),
```

This way we don't create and destroy arrays unnecessarily... Or am I missing some context?